concordium-std-derive/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+extern crate proc_macro;
+
+use concordium_contracts_common::{
+    AccountAddress, ContractAddress, ModuleReference, PublicKeyEcdsaSecp256k1, PublicKeyEd25519,
+    SignatureEcdsaSecp256k1, SignatureEd25519,
+};
+use proc_macro::TokenStream;
+use std::str::FromStr;
+use syn::{parse_macro_input, LitStr};
+
+/// Procedural macro for instantiating account addresses.
+/// Input must be a valid base58-encoding.
+#[proc_macro]
+pub fn account_address(item: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(item as LitStr);
```

Comment:
This also correctly handles the case where we pass in anything that is _not_ a string literal, fx:

```rust
const ACC: AccountAddress = account_address!(0);
```

Will correctly be caught, and produce the following error:

```
error: Invalid account address: Invalid Base58Check encoding.
 --> tests/test-programs/fail-acc-invalid-base58.rs:6:46
  |
6 | const ACC: AccountAddress = account_address!("30Bx2h5Y2veb4hZgAJWPrr8RyQESKm5TjzF3ti1QQ4VSYLwK1G");
  |                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
That does not look like the right error. We want to tell the user they need a string literal.